config/src/test/java/io/confluent/common/config/ConfigDefTest.java

```diff
@@ -275,4 +285,79 @@ private void testValidators(Type type, ConfigDef.Validator validator, Object def
       }
     }
   }
+
+  @Test
+  public void testClassLoading() throws IOException {
+    ClassLoader systemLoader = ClassLoader.getSystemClassLoader();
+
+    // test with system classloader
+    String configName = "type.class.property";
+    String configNameDefault = Password.class.getName();
+
+    ConfigDef configDef = new ConfigDef();
+    configDef.define(configName, Type.CLASS, configNameDefault,
+        ConfigDef.Importance.MEDIUM, "system class");
+
+    AbstractConfig config = new AbstractConfig(configDef, new HashMap<>());
+
+    Class<?> systemKlass = config.getClass(configName);
+    assertEquals(systemLoader, systemKlass.getClassLoader());
+
+    // test with custom, child-first, classloader
+    Set<String> allowedKlasses = Collections.singleton(configNameDefault);
+    List<URL> pluginUrls = Collections.singletonList(Paths.get("target/classes/").toUri().toURL());
+    URLClassLoader pluginLoader = new PluginClassLoader(pluginUrls.toArray(new URL[0]), systemLoader, allowedKlasses);
+
+    Thread.currentThread().setContextClassLoader(pluginLoader);
+
+    configDef = new ConfigDef();
+    configDef.define(configName, Type.CLASS, configNameDefault, ConfigDef.Importance.MEDIUM, "plugin class");
```

Comment:
I think you can re-use the same ConfigDef object, and just have the AbstractConfig constructed with the alternate classloader.

Reply:
I construct the `AbstractConfig` in order to get the value back. My test as intended tests `parseType` inside `define`. I don't see a good reason to add a redirection here. Also, I think I would have to add the type in the map in order to call `parseType` during `AbstractConfig`'s creation. Again, given that I don't want to change visibility of methods in this PR, I'd like to keep this test as close as possible to a unit test of the parsing.

Reply:
Oh I see, the critical parseType is actually in `define` in the confluent ConfigDef, which is different than AK's ConfigDef. This makes my earlier suggestion invalid.